tests/system/chatbot/test_chatbot.py

```diff
@@ -1,48 +1,67 @@
 import pytest
-import requests
-from xprocess import ProcessStarter
 
+from jina import Flow, Document
+from jina.helloworld.chatbot.app import hello_world
 from jina.parsers.helloworld import set_hw_chatbot_parser
+from jina.helloworld.chatbot.my_executors import MyTransformer, MyIndexer
+from tests import validate_callback
 
 
 @pytest.fixture
-def chatbot_args(tmpdir):
+def helloworld_args(tmpdir):
     return set_hw_chatbot_parser().parse_args(
-        ['--workdir', str(tmpdir), '--port-expose', '8080']
+        [
+            '--workdir',
+            str(tmpdir),
+            '--port-expose',
+            '8080',
+            '--unblock-query-flow',
+            '--parallel',
+            '1',
+        ]
     )
 
 
-@pytest.fixture
-def payload():
-    return {'top_k': 1, 'data': ['text:Is my dog safe from virus']}
-
-
-@pytest.fixture
-def post_uri():
-    return 'http://localhost:8080/search'
+def test_chatbot_helloworld(helloworld_args):
+    hello_world(helloworld_args)  # ensure no error is raised.
 
 
-@pytest.fixture
-def expected_result():
-    return '''It's not completely up to you.'''
-
+def test_chatbot(tmpdir, mocker):
+    #  test the index and query flow.
+    def validate_response(resp):
+        assert len(resp.data.docs) == 1
+        for doc in resp.data.docs:
+            assert len(doc.matches) == 1
+            assert (
+                'testing positive means that you have the virus'
+                in doc.matches[0].tags['answer']
+            )
 
-@pytest.fixture(autouse=True)
-def start_server(xprocess, chatbot_args):
-    class Starter(ProcessStarter):
-        pattern = "You should see a demo page opened in your browser"
-        args = ["jina", "hello", "chatbot"]
-        max_read_lines = 10000
-
-    xprocess.ensure("server", Starter)
-    yield
-    xprocess.getinfo("server").terminate()
+    mock_on_done = mocker.Mock()
+    mock_on_fail = mocker.Mock()
 
+    flow = (
```

Comment:
to make sure that we use the `exact` same `flow`, we could add a function in the helloworld that creates the `chatbot flow`? Like this we can make sure that changes in the Flow are properly tested